.azuredevops/pipelineTemplates/module.jobs.deploy.yml

```diff
@@ -0,0 +1,287 @@
+                }
+                # Invoke Token Replacement Functionality
+                $null = Convert-TokensInParameterFile @ConvertTokensInputs -Verbose
+
+          # [Validation] task(s)
+          #---------------------
+          - task: AzurePowerShell@5
+            displayName: 'Validate [${{ deploymentBlock.path }}] via connection [${{ parameters.serviceConnection }}]'
+            inputs:
+              azureSubscription: ${{ parameters.serviceConnection }}
+              azurePowerShellVersion: ${{ parameters.azurePowerShellVersion }}
+              preferredAzurePowerShellVersion: ${{ parameters.preferredAzurePowerShellVersion }}
+              ScriptType: InlineScript
+              pwsh: true
+              inline: |
+                # Load used functions
+                . (Join-Path '$(moduleRepoRoot)' '$(pipelineFunctionsPath)' 'resourceValidation' 'Test-TemplateWithParameterFile.ps1')
+
+                $templateFilePath = ([String]::IsNullOrEmpty('${{ deploymentBlock.templateFilePath }}')) ?
+                  (Join-Path '$(ModuleRepoRoot)' '$(modulePath)' 'deploy.bicep') :
+                  (Join-Path '$(moduleRepoRoot)' '${{ deploymentBlock.templateFilePath }}')
+
+                # ----------- #
+                # INVOKE TEST #
+                # ----------- #
+                $functionInput = @{
+                  templateFilePath   = $templateFilePath
+                  parameterFilePath  = Join-Path '$(ParametersRepoRoot)' '${{ deploymentBlock.path }}'
+                  location           = '${{ parameters.location }}'
+                  resourceGroupName  = '${{ parameters.resourceGroupName }}'
+                  subscriptionId     = '${{ parameters.subscriptionId }}'
+                  managementGroupId  = '${{ parameters.managementGroupId }}'
+                }
+
+                Write-Verbose "Invoke task with" -Verbose
+                Write-Verbose ($functionInput | ConvertTo-Json | Out-String) -Verbose
+
+                Test-TemplateWithParameterFile @functionInput -Verbose
+
+          # [Deployment] task(s)
+          #---------------------
+          - task: AzurePowerShell@5
+            displayName: 'Deploy [${{ deploymentBlock.path }}] via connection [${{ parameters.serviceConnection }}]'
+            inputs:
+              azureSubscription: ${{ parameters.serviceConnection }}
+              azurePowerShellVersion: ${{ parameters.azurePowerShellVersion }}
+              preferredAzurePowerShellVersion: ${{ parameters.preferredAzurePowerShellVersion }}
+              pwsh: true
+              ScriptType: InlineScript
+              inline: |
+                # Load used functions
+                . (Join-Path '$(moduleRepoRoot)' '$(pipelineFunctionsPath)' 'resourceDeployment' 'New-ModuleDeployment.ps1')
+
+                $templateFilePath = ([String]::IsNullOrEmpty('${{ deploymentBlock.templateFilePath }}')) ?
+                  (Join-Path '$(ModuleRepoRoot)' '$(modulePath)' 'deploy.bicep') :
```

Comment:
Can we make this more interoperable with json too?

Reply:
done